api/v1beta1/azuremanagedcontrolplane_webhook.go

```diff
@@ -492,6 +497,35 @@ func (m *AzureManagedControlPlane) validateManagedClusterNetwork(cli client.Clie
 	return nil
 }
 
+// validateAutoUpgradeProfile validates auto upgrade profile.
+func (m *AzureManagedControlPlane) validateAutoUpgradeProfile(old *AzureManagedControlPlane) field.ErrorList {
+	var allErrs field.ErrorList
+	if old.Spec.AutoUpgradeProfile != nil && m.Spec.AutoUpgradeProfile == nil {
+		// Prevent AutoUpgradeProfile to be set to nil.
+		// Unsetting the field is not allowed.
+		allErrs = append(allErrs,
+			field.Invalid(
+				field.NewPath("Spec", "AutoUpgradeProfile"),
+				m.Spec.AutoUpgradeProfile,
+				"field cannot be set to nil, to disable auto upgrades set the channel to none."))
+	}
+
```

Comment:
@LochanRn Actually, I think a better approach would be to have `validateCreate()` above do a check on `m.Spec.Version` like this. Since the field is not optional, we shouldn't let the set the field to that value at all if won't parse.

```
_, err := semverv4.ParseTolerant(m.Spec.Version)
if err != nil {
	return errors.new("version is invalid", m.Spec.Version)
}
```

Then, in validateUpdate, we can just validate the new value `m.Spec.Version`, and since we know they're both valid, go ahead and use `GTE()` directly instead of needing a helper function.

```
newVersion, err := semverv4.ParseTolerant(m.Spec.Version)
if err != nil {
	return errors.new("version is invalid", m.Spec.Version)
}
oldVersion, err := semverv4.ParseTolerant(old.Spec.Version)
if err != nil {
	// Note: we don't expect this to fail as we already validated the version
	return errors.new("version is invalid", old.Spec.Version)
}
if newVersion.GTE(oldVersion) {
	...
}
```

Reply:
I updated the helper funtcion to not return any error and in validate create i updated the code to parse the version for semver4.ParseTolerant.